spec/handlers/sessions_create_spec.rb

```diff
@@ -356,6 +356,70 @@
             end
           end
         end
+
+        context "linked to no one" do
+          context "but user already has already authenticated with the same provider before" do
+            context "new_social user" do
+              let!(:user) { FactoryGirl.create(:new_social_user) }
+              let!(:authentication) { FactoryGirl.create(:authentication, user: user) }
+
+              it "returns a status output instead of raising an exception" do
+                result = lambda { |uid|
```

Comment:
Minor, but... The above `result`s are true handler results.  I think this needs a different name, because it isn't a "result" but a callable thing that produces a result.  `handler_wrapper` or something